doc/unordered/hash_traits.adoc

```diff
@@ -32,9 +32,14 @@ large changes in the returned hash code &#8212;ideally, flipping one bit in the
 the input value results in each bit of the hash code flipping with probability 50%. Approaching
 this property is critical for the proper behavior of open-addressing hash containers.
 
-`hash_is_avalanching<Hash>::value` is `true` if `Hash::is_avalanching` is a valid type,
-and `false` otherwise.
-Users can then declare a hash function `Hash` as avalanching either by embedding an `is_avalanching` typedef
+`hash_is_avalanching<Hash>::value` is:
+
+ * `false` if `Hash::is_avalanching` is not present,
+ * `Hash::is_avalanching::value` if this is present and convertible at compile time to a `bool`,
+ * `true` if `Hash::is_avalanching` is `void` (this usage is deprecated).
+ 
+The behavior is undefined if none of the three cases above is met.
```

Comment:
I don't think we want "undefined" here.

Reply:
You're right, I miswrote that for "ill-formed". What's specifically ill formed? The "program", the expression `hash_is_avalanching<Hash>::value`, something else?

Reply:
The program, I suppose.

Reply:
I think I'll go with "the expression", as the user can legitimately define and use something like:
```cpp
struct my_hash{ using is_avalanching=int; };
```
as long as he doesn't plug into Boost.Unordered.